### PR TITLE
also build shared library in recent HDF easyconfigs

### PR DESCRIPTION
**Pull request by verdurin**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by akesandgren** [comment]
Why would you need to disable fortran just to make shared libs?

**Comment by verdurin** [comment]
https://portal.hdfgroup.org/display/knowledge/How+do+I+install+HDF4:
"Fortran shared libraries are not supported. Shared libraries are disabled by default."

**Comment by boegel** [comment]
@verdurin Why not make `configopts` a list of values, and first installed shared libs only, then install static libs (and retain Fortran support)?

**Comment by verdurin** [comment]
@boegel do you mean effectively to build it twice? Would making the list of `configopts` be sufficient to achieve that?

Otherwise I don't understand.

**Comment by boegel** [comment]
> @boegel do you mean effectively to build it twice? Would making the list of `configopts` be sufficient to achieve that?
> 
> Otherwise I don't understand.

Yes, basically building it twice. Making `configopts` a list is sufficient, for every entry in `configopts` the `configure`-`build`-`test`-`install` cycle will be run multiple times.

**Comment by verdurin** [comment]
Worth highlighting that in order to work with `GDAL` I needed to disable `netcdf` support.

File: easybuild/easyconfigs/h/HDF/HDF-4.2.14-GCCcore-8.3.0.eb [hunk 1]
```diff
@@ -31,14 +31,21 @@ dependencies = [
...
-configopts += '--includedir=%(installdir)s/include/%(namelower)s '
+configopts = [
+    '--with-szlib=$EBROOTSZIP CFLAGS="$CFLAGS -I$EBROOTLIBTIRPC/include/tirpc" \
+    --includedir=%(installdir)s/include/%(namelower)s ',
+    '--with-szlib=$EBROOTSZIP CFLAGS="$CFLAGS -I$EBROOTLIBTIRPC/include/tirpc" \
+    --includedir=%(installdir)s/include/%(namelower)s \
+    --enable-shared --disable-fortran --disable-netcdf',  # Cannot build shared libraries and Fortran...
+    # https://trac.osgeo.org/gdal/wiki/HDF#IncompatibilitywithNetCDFLibraries
+    # netcdf must be disabled to allow HDF to be used by GDAL
+]
```
Comment (smoors) [inline easybuild/easyconfigs/h/HDF/HDF-4.2.14-GCCcore-8.3.0.eb:42]: you can avoid the duplicated lines like this:

```suggestion
_configopts = '--with-szlib=$EBROOTSZIP CFLAGS="$CFLAGS -I$EBROOTLIBTIRPC/include/tirpc" '
_configopts += '--includedir=%(installdir)s/include/%(namelower)s '
configopts = [
    _configopts,
    # Cannot build shared libraries and Fortran:
    # https://trac.osgeo.org/gdal/wiki/HDF#IncompatibilitywithNetCDFLibraries
    # netcdf must be disabled to allow HDF to be used by GDAL
    _configopts + '--enable-shared --disable-fortran --disable-netcdf',
]
```

**Review by smoors — COMMENTED** [review]
@verdurin I'm a bit worried that software depending on both HDF and netCDF will now pick up the shared libs instead of the static libs, and thus start to fail.

maybe it's safer to create a new easyconfig, to be used by GDAL, with a versionsuffix? for example '-nonetcdf'?

File: easybuild/easyconfigs/h/HDF/HDF-4.2.14-GCCcore-8.3.0.eb (hunk 1 above)
Comment (zao) [reply easybuild/easyconfigs/h/HDF/HDF-4.2.14-GCCcore-8.3.0.eb:42]: I've seen `local_*` as the prefix for things that shouldn't be interpreted by EasyBuild in the past, does `_*` have a similar treatment and if so, which should be preferred when?

**Comment by zao** [comment]
Test report by @zao
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
eb-u1604 - Linux ubuntu 16.04, x86_64, Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz, Python 2.7.12
See https://gist.github.com/6675d33fc8b98572c48c604b86aac516 for a full test report.

File: easybuild/easyconfigs/h/HDF/HDF-4.2.14-GCCcore-8.3.0.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/h/HDF/HDF-4.2.14-GCCcore-8.3.0.eb:42]: Both `local_*` and `_*` are valid for local variables, see https://docs.easybuild.io/en/latest/Easyconfig-files-local-variables.html#recommended-naming-scheme-for-local-variables-in-easyconfig-files

**Comment by boegel** [comment]
> @verdurin I'm a bit worried that software depending on both HDF and netCDF will now pick up the shared libs instead of the static libs, and thus start to fail.
> 
> maybe it's safer to create a new easyconfig, to be used by GDAL, with a versionsuffix? for example '-nonetcdf'?

That's only really relevant for `NCL` in the easyconfigs we have in the central repo, so easy to test?

Also, I'm not sure anything would actually get broken, since it could just pick up shared libs for HDF and static libs for netCDF, and it should still work fine?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3501.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/a6d1fe8455b69c83d975893f04333a02 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node2632.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/391724e71b6d6862efec1bf07dc43d9c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3155.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/0d1db0a7da0d33e18b8f0b301fd018c0 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Going in, thanks @verdurin!

**Pull request merged**